### PR TITLE
Add initial tests for analyzer.go

### LINKED ISSUE #42: Create a test file for analyzer.go
It is not necessary to cover all code elements. Just a few methods.
Source file: https://github.com/rodrigo-brito/gocity/blob/master/analyzer/analyzer.go

### PR DESCRIPTION
**Pull request by alebcay**
This PR fixes #42. I've tried to cover a number of cases, especially on `TestAnalyze`, open to suggestions if there's anything that should be added/omitted.

### CODE REVIEW

File: pkg/analyzer/analyzer_test.go [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+
+	_, err = f.WriteString("package main\nimport \"fmt\"\nfunc main(){fmt.Println(\"hello world\")}\n")
+
+	expectedSimple := &NodeInfo{NumberLines: 1, NumberMethods: 1}
+	return expectedSimple, err
+}
```
Comment (rodrigo-brito) [inline pkg/analyzer/analyzer_test.go:168]: you can use a similar file in` testdata`

File: pkg/analyzer/analyzer_test.go [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+func (r *rect) area() int { return r.w * r.h }
+func (r *rect) perim() int { return 2*r.w + 2*r.h }
+func main() {
+    r := rect{w: 10, h: 5}
+    fmt.Println("area: ", r.area())
+    fmt.Println("perim:", r.perim())
+}`
+
+	os.MkdirAll(dir, 0755)
+	f, err := os.Create(filepath.Join(dir, "main.go"))
```
Comment (rodrigo-brito) [inline pkg/analyzer/analyzer_test.go:183]: The same here

**Review by rodrigo-brito — CHANGES_REQUESTED** [review]
Thank you so much for the help, just a couple of comments

**Comment by alebcay** [comment]
Thanks, I did not realize there was already a test package available. I've rewritten the test to work with the existing files rather than adding new ones. Please advise if you would prefer me to add new files into the test package instead of using the existing files.

**Review by rodrigo-brito — APPROVED** [review]

**Comment by rodrigo-brito** [comment]
Gracias @alebcay

**Pull request merged**